server/src/main/java/org/apache/cloudstack/backup/BackupManagerImpl.java

```diff
@@ -585,17 +595,64 @@ public boolean restoreBackup(final Long backupId) {
 
         final BackupOffering offering = backupOfferingDao.findByIdIncludingRemoved(vm.getBackupOfferingId());
         if (offering == null) {
-            throw new CloudRuntimeException("Failed to find backup offering of the VM backup");
+            throw new CloudRuntimeException("Failed to find backup offering of the VM backup.");
         }
 
-        final BackupProvider backupProvider = getBackupProvider(offering.getProvider());
-        if (!backupProvider.restoreVMFromBackup(vm, backup)) {
-            throw new CloudRuntimeException("Error restoring VM from backup ID " + backup.getId());
+        String backupDetailsInMessage = ReflectionToStringBuilderUtils.reflectOnlySelectedFields(backup, "uuid", "externalId", "vmId", "type", "status", "date");
+        try {
+            updateVmState(vm, VirtualMachine.Event.RestoringRequested);
+            updateVolumeState(vm, Volume.Event.RestoreRequested);
+            final BackupProvider backupProvider = getBackupProvider(offering.getProvider());
+            if (!backupProvider.restoreVMFromBackup(vm, backup)) {
+                throw new CloudRuntimeException(String.format("Error restoring %s from backup [%s].", vm, backupDetailsInMessage));
+            }
+        } catch (Exception e) {
```

Comment:
if this catch-all is really necessary, please comment on that.

Reply:
@DaanHoogland has the restore process is made by an external provider, I chose to put a catch-all to ensure that no provider-side exception could be missed. Please let me know if my explanation wasn't enough (communication is not my strong point :()

Reply:
Your explanation is perfect, @SadiJr better than any I have ever heard about a pokemon catch. I would like to see it in a comment in the code, so people that don't see this PR know the reason as well ;)

Reply:
Done, thanks for this suggestions, I hadn't thought of that possibility.